packages/core/strapi/lib/services/entity-service/types/params/publication-state.d.ts

```diff
@@ -8,7 +8,15 @@ export type IsEnabled<TSchemaUID extends Common.UID.Schema> = Utils.Expression.M
       Common.UID.IsContentType<TSchemaUID>,
       Utils.Expression.IsTrue<Common.Schemas[TSchemaUID]['options']['draftAndPublish']>
     ],
-    [Common.UID.IsComponent<TSchemaUID>, Utils.Expression.False]
+    [
+      // Here, we're manually excluding potential overlap between Component and ContentTypes' UIDs and thus preventing false positives
+      // e.g. api::foo.bar extends a Component UID (`${string}.${string}`) but shouldn't be considered a component
+      Utils.Expression.And<
```

Comment:
If we can directly make the Component UID not match the ContentType one that will void having to drop this expression in many places later.